pathd/path_zebra.c

```diff
@@ -155,32 +336,125 @@ static int path_zebra_router_id_update(ZAPI_CALLBACK_ARGS)
 	return 0;
 }
 
+/**
+ * Disconnect from NHT
+ */
+void path_nht_removed(struct srte_candidate *candidate)
+{
+	struct path_nht_data *nhtd, lookup;
+	struct srte_segment_list *segment_list;
+	bool was_zebra_registered;
+
+	if (!candidate || !candidate->lsp)
+		return;
+
+	segment_list = candidate->lsp->segment_list;
+	if (!segment_list)
+		return;
+
+	if (!CHECK_FLAG(segment_list->flags, F_SEGMENT_LIST_NHT_REGISTERED))
+		return;
+
+	if (!path_zebra_nht_get_srv6_prefix(segment_list, &lookup.nh))
+		return;
+
+	lookup.nh_vrf_id = VRF_DEFAULT;
+
+	/* nh->nh_registered means we own a reference on the nhtd */
+	nhtd = path_nht_hash_find(path_nht_hash, &lookup);
+
+	assertf(nhtd, "BUG: NH %pFX registered but not in hashtable",
+		&lookup.nh);
+
+	was_zebra_registered = nhtd->registered;
+	UNSET_FLAG(segment_list->flags, F_SEGMENT_LIST_NHT_REGISTERED);
+	if (path_nht_hash_decref(&nhtd))
+		/* still got references alive */
+		return;
+
+	/* NB: nhtd is now NULL. */
+	if (!was_zebra_registered)
+		return;
+
+	zlog_debug("Unregistering nexthop(%pFX) for candidate %s pref %u",
```

Comment:
Should be guarded by debug flag?

Reply:
could you please elaborate? It seems like other parts of code do it in similar way? https://github.com/FRRouting/frr/blob/57fdef4550c601fc84be1f2a04e591ea6e19f381/pathd/path_zebra.c#L672
https://github.com/FRRouting/frr/blob/57fdef4550c601fc84be1f2a04e591ea6e19f381/pathd/path_zebra.c#L692

Reply:
In a future pull request, a 'debug pathd nth' extension command should be planned.

Reply:
yeah this is a blocker from my perspective adding more unguarded debugs to a PR, even if the original code snuck some in, is a no go.

Reply:
remove the blocker -> fixed.